### PR TITLE
feat: enable osnap selection to restrict voting types

### PR DESCRIPTION
**Pull request by daywiss**
### Issues
Its been suggested that we restrict voting type to "basic" when doing an osnap vote. This is because osnap team does not have the ability to automatically monitor and dispute arbitrary vote types where user may define other fields. Restricting this will allow more robust on chain monitoring and disputing of bad proposals.

Fixes #

### Changes 
1. we add code to check if the user has a valid osnap install
2. if they do, during proposal building, we add a selection where the user can specify if they want to use osnap 
3. if they choose yes, we hard code the vote type to basic, and prevent user from selecting anything else

this only shows if osnap is enabled and working:
image

### How to test
1. go to /#/outcomefinance.eth/create, which should be an osnap space
2. begin the proposal process, see that you are asked if you want to make this an osnap vote. 
3. select yes, see that in the next screen you are limited to basic vote.
4. go to next page. see that you can add transaction information
5. go back to first page, select no.
6. go to next page, see that you are free to select any type of vote
7. see that on vote page, you can only plublish, skipping transaction page.
8. go to a non osnap space that you can propose on
9. create new proposal and see that there is no osnap related messaging, and all vote types are enabled.

### Self-review checklist
- [x] I have performed a full self-review of my changes
- [x] I have tested my changes on a preview deployment
- [ ] I have tested my changes on different screen sizes (sm, md)
- [ ] I have tested my changes on a custom domain

### Additional notes or considerations
*_(Include any other relevant information or context that may be helpful for the reviewer)_*

### CODE REVIEW

**Comment by daywiss** [comment]
thanks for vercel approval, let me get some testers on here before we merge

**Comment by daywiss** [comment]
need another vercel approval, updated an issue

**Comment by samuveth** [comment]
Thanks! We will review soon

**Comment by daywiss** [comment]
we have approval from our internal testing

**Comment by bonustrack** [comment]
@daywiss Thanks! It seem to work well, I think for the UX it will be better if the message appear on the second step, as the first step is not related to execution. I would suggest to just show a message saying execution with OSnap requires basic voting type under the voting type selection without select list.

**Comment by bonustrack** [comment]
@hlambur Got it. Maybe instead of the selector we could have a checkbox just before the voting type selector saying "This proposal will include transaction to execute with oSnap" and enforce Basic type if this checkbox is selected.

**Comment by hlambur** [comment]
@bonustrack I like the checkbox! You want that checkbox to live on the second screen (just above the voting type selector)? If so we can have @daywiss or @ryanwolhuter assist with that.

**Comment by bonustrack** [comment]
@hlambur Correct! That'd be fantastic

**Comment by daywiss** [comment]
ok made the changes to move this to the voting page, and turn it into a checkbox
image

I've tested it and seems to work correctly, ie it will disable other voting types, and enable transaction page if enabled. if not enabled, all voting types are avaialble, and transaction page is skipped. this will also maintain state between going back and forth between stages in proposal flow.

File: src/components/SpaceCreateVoting.vue
Comment (bonustrack) [inline src/components/SpaceCreateVoting.vue:101]: We should use the official casing here
```suggestion
      Yes, use oSnap for transactions (this will restrict voting types).
```

File: src/components/SpaceCreateVoting.vue
Comment (bonustrack) [inline src/components/SpaceCreateVoting.vue:91]:
```suggestion
      Are you planning for this proposal to initiate a transaction that your
      organization Safe will execute if approved? (Remember, oSnap enables
      trustless and permissionless execution)
```

File: src/components/SpaceCreateVoting.vue
Comment (bonustrack) [inline src/components/SpaceCreateVoting.vue:104]: It's better to use tailwind class like mb-4 instead of multiple `<br/>` here

File: src/views/SpaceCreate.vue
Comment (bonustrack) [inline src/views/SpaceCreate.vue:8]: Imports should be separated from others code
```suggestion
import Plugin from '@/plugins/safeSnap';

const safeSnapPlugin = new Plugin();
```

File: src/components/SpaceCreateVoting.vue
Comment (bonustrack) [inline src/components/SpaceCreateVoting.vue:87]:
```suggestion
    <h6>oSnap proposal</h6>
```

**Comment by bonustrack** [comment]
Ideally if someone check for oSnap we should enforce that he can't select others voting type, currently it's possible to select oSnap but change voting type to something else.

**Review by bonustrack — APPROVED** [review]
Looks good! tACK

**Pull request merged**
